src/nrtk_explorer/library/object_detector.py

```diff
@@ -93,12 +94,12 @@ def eval(
             for group in images.values()
         ]
         # Flatten the list of predictions
-        predictions = reduce(operator.iadd, predictions)  # type: ignore
-
-        output = list()
-        for path in paths:
-            for prediction in predictions:
-                if prediction[0] == path:
-                    output.append(prediction)
-
-        return output
+        predictions = reduce(operator.iadd, predictions, [])
+
+        # order output by paths order
+        find_prediction = lambda id: next(
+            prediction for prediction in predictions if prediction[0] == id
+        )
+        output = [find_prediction(id) for id in image_ids]
```

Comment:
Looks like standard Functional Programming to me. If we don't like that, then we should be flattening our 2D lists with 2 for loops rather than `reduce`.  With the old code, it read like a compare all to all algorthim with the double for loop.

Reply:
I'll change this to make an ID to Prediction map, then a reorder pass.

Reply:
Decided to just return the ID to Predictions Dict, and tweek the downstream code.
```
        predictions_by_image_id = {
            image_id: predictions
            for batch in predictions_in_baches
            for image_id, predictions in batch
        }
        return predictions_by_image_id
```

Reply:
> Looks like standard Functional Programming to me.  If we don't like that, then we should be flattening our 2D lists with 2 for loops rather than reduce. 

My comment was about readability which is of course a subjective topic. I consider that using a lambda function defined lines above takes a couple of seconds more of making sense than the conditional inside the for loop.

> Decided to just return the ID to Predictions Dict, and tweek the downstream code.

@PaulHax The idea is that we pass a list of ids and we are returned a list of the annotations for each of this id in the order of the ids.

Lets take this refactor to another PR where we can discuss this.